www/src/components/Breadcrumbs.js

```diff
@@ -75,9 +77,10 @@ export function Breadcrumbs() {
 
 export default function BreadcrumbProvider({ children }) {
   const [breadcrumbs, setBreadcrumbs] = useState([])
+  const value = useMemo(() => ({ breadcrumbs, setBreadcrumbs }), [breadcrumbs])
```

Comment:
These useMemo prevent a lot of useless rerendering of the application. It makes it way faster hence.

Reply:
very helpful